R/generate_report.R

```diff
@@ -0,0 +1,79 @@
+#' Generate a report for a plate.
+#'
+#' This function generates a report for a plate. The report is generated using
+#' the `plate_report_template.Rmd` template.
+#'
+#' @param plate A plate object.
+#' @param use_model (`logical(1)`) A logical value indicating whether the model should be used in the report.
+#' @param filename (`character(1)`) The name of the output file. If not provided,the filename will be created based on the plate name with the suffix '_report.html'.
+#' @param output_dir (`character(1)`) The directory where the report should be saved. Default is 'reports'.
+#' @param counts_lower_threshold (`numeric(1)`) The lower threshold for the counts plots (works for each analyte). Default is 50.
+#' @param counts_higher_threshold (`numeric(1)`) The higher threshold for the counts plots (works for each analyte). Default is 70.
+#'
+#'
+#' @return A report.
+#'
+#' @examples
+#'
+#' plate_file <- system.file("extdata", "CovidOISExPONTENT.csv", package = "PvSTATEM")
+#' layout_file <- system.file("extdata", "CovidOISExPONTENT_layout.csv", package = "PvSTATEM")
+#'
+#' plate <- read_luminex_data(plate_file, layout_file)
+#' generate_plate_report(plate, counts_lower_threshold = 40, counts_higher_threshold = 50)
+#' @export
+generate_plate_report <- function(plate, use_model = TRUE, filename = NULL, output_dir = "reports", counts_lower_threshold = 50, counts_higher_threshold = 70) {
```

Comment:
I would also add option to create custom output report filename - to keep this consistent with `process_plate` function. I am not sure, which approach is preferred by Michael, this is to be consulted on Friday, but I would stick to just one, consistent across all the methods

Reply:
this option does not need to be added necessarily in this release, I am mentioning it here so that it won't be forgotten

Reply:
Hmm, it's true that implementation of the custom file name is handled differently there and in `process_plate`. In future releases, we should focus on unifying this. Here we provide seperatly custom file name and path to the directory, and in `process_plate` it's done in one argument. Also, the defaults for output-dir are different, and this needs to be addressed asap. In the `process_plate` default is the current working directory, and here it's /reports. At the moment, I don't know which one is better and how CRAN will respond to the latter.